python_modules/dagster/dagster_tests/asset_defs_tests/test_partitioned_assets.py

```diff
@@ -484,6 +494,78 @@ def my_asset():
     )
 
 
+def test_multi_asset_with_different_partitions_defs():
```

Comment:
Can you add a test case with `can_subset=False` and all specs having the same partition def? I think this should be allowed.